libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2463,9 +2463,16 @@ string YulUtilFunctions::cleanupFromStorageFunction(Type const& _type, bool _spl
 				return templ.render();
 			}
 
+		bool leftAligned = false;
+		if (
+			_type.category() != Type::Category::Function ||
+			dynamic_cast<FunctionType const&>(_type).kind() == FunctionType::Kind::External
+		)
+			leftAligned = _type.leftAligned();
```

Comment:
So external function variables are left aligned because it's the signature? And internal ones are not because...?

Reply:
External functions have two encodings:
 - in memory, storage and calldata they are "<address><signature>" encoded as a `bytes24`
 - on the stack, it is just two different variables, address is encoded as an `address` type and `signature` is encoded as a `bytes4` (or maybe `uint32`, I don't know)

But external functions on the stack (and actually in general) are not supported by this PR, this is what the boolean parameter is for.

Reply:
Ok